traits/ctraits.c

```diff
@@ -3677,11 +3677,31 @@ validate_trait_callable(
     trait_object *trait, has_traits_object *obj, PyObject *name,
     PyObject *value)
 {
-
-    if ((value == Py_None) || PyCallable_Check(value)) {
+    if (PyCallable_Check(value)) {
         Py_INCREF(value);
         return value;
     }
+    else if (value == Py_None) {
+
+        int tuple_size = PyTuple_GET_SIZE(trait->py_validate);
+
+        //Handle callables without allow_none, default to allow None
+        if(tuple_size < 2){
+            Py_INCREF(value);
+            return value;
+        }
+
+        int allow_none = PyObject_IsTrue(PyTuple_GET_ITEM(trait->py_validate, 1));
+
+        if (allow_none == -1) {
+            return NULL;
+        }
+
+        else if(allow_none) {
```

Comment:
Style nitpick. When in doubt, match the style to existing code.
```suggestion
        else if (allow_none) {
```